packages/@aws-cdk/aws-cloudwatch/lib/query.ts

```diff
@@ -0,0 +1,75 @@
+import * as cdk from '@aws-cdk/core';
+import { ConcreteWidget } from "./widget";
+
+/**
+ * Properties for a Query widget
+ */
+export interface QueryWidgetProps {
+  /**
+   * Title for the graph
+   *
+   * @default No title
+   */
+  readonly title?: string;
+
+  /**
+   * Log group to query
+   */
+  readonly logGroup: string;
+
+  /**
+   * Query for log insights
+   */
+  readonly query: string;
+
+  /**
+   * The region the metrics of this graph should be taken from
+   *
+   * @default Current region
+   */
+  readonly region?: string;
+
+  /**
+   * Width of the widget, in a grid of 24 units wide
+   *
+   * @default 6
+   */
+  readonly width?: number;
+
+  /**
+   * Height of the widget
+   *
+   * @default 6
+   */
+  readonly height?: number;
+}
+
+/**
+ * Display query results from Logs Insights
+ */
+export class QueryWidget extends ConcreteWidget {
+  private readonly props: QueryWidgetProps;
+
+  constructor(props: QueryWidgetProps) {
+    super(props.width || 6, props.height || 6);
+    this.props = props;
+  }
+
+  public toJson(): any[] {
+    return [{
+      type: 'log',
+      width: this.width,
+      height: this.height,
+      x: this.x,
+      y: this.y,
+      properties: {
+        view: 'table',
+        title: this.props.title,
+        region: this.props.region || cdk.Aws.REGION,
+        query: cdk.Fn.join('', [
```

Comment:
You can use a string substitution here using backticks:

```
`SOURCE '${this.logGroup.logGroupName}' | ${this.props.query}`
```

Reply:
Does it need to be the log group name or the ARN, btw?

Reply:
Can it be in a different account?

Reply:
It needs to be the name. I haven't seen any way to do it cross account.